File: t/PACSFramework/CS/DicomHighlevelPatientUpdaterDemo/Dicom/ChangeSeries.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Leadtools.Dicom.Common.Attributes;
using Leadtools.Dicom;
using Leadtools.Dicom.Common.Extensions;
using System.ComponentModel;
using Leadtools.Dicom.Common.DataTypes;

namespace DicomDemo.Dicom
{
    [Instance(DicomClassType.Undefined, NActionScu.PatientUpdateClass)]
    public class ChangeSeries : PatientUpdate
    {
        private DateTime? _SeriesDate;

        [Element(DicomTag.SeriesDate)]        
        public DateTime? SeriesDate
        {
            get { return _SeriesDate; }
            set { _SeriesDate = value; }
        }

        private DateTime? _SeriesTime;

        [Element(DicomTag.SeriesTime)]
        public DateTime? SeriesTime
        {
            get { return _SeriesTime; }
            set { _SeriesTime = value; }
        }

        private string _SeriesDescription;

        [Element(DicomTag.SeriesDescription)]
        public string SeriesDescription
        {
            get { return _SeriesDescription; }
            set { _SeriesDescription = value; }
        }

        private string _Modality;

        [Element(DicomTag.Modality)]
        public string Modality
        {
            get { return _Modality; }
            set { _Modality = value; }
        }        

        private string _SeriesInstanceUID;

        [Element(DicomTag.SeriesInstanceUID)]
        public string SeriesInstanceUID
        {
            get { return _SeriesInstanceUID; }
            set { _SeriesInstanceUID = value; }
        }
    }
}

File: t/PACSFramework/VB/DicomHighlevelPatientUpdaterDemo/Dicom/ChangeSeries.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.ComponentModel
Imports Leadtools.Dicom.Common.Attributes
Imports Leadtools.Dicom
Imports Leadtools.Dicom.Common.Extensions

Namespace DicomDemo.Dicom
   <Instance(DicomClassType.Undefined, NActionScu.PatientUpdateClass)> _
   Public Class ChangeSeries : Inherits PatientUpdate
      Private _SeriesDate As Nullable(Of DateTime)

      <Element(DicomTag.SeriesDate)> _
      Public Property SeriesDate() As Nullable(Of DateTime)
         Get
            Return _SeriesDate
         End Get
         Set(ByVal value As Nullable(Of DateTime))
            _SeriesDate = Value
         End Set
      End Property

      Private _SeriesDescription As String

      <Element(DicomTag.SeriesDescription)> _
      Public Property SeriesDescription() As String
         Get
            Return _SeriesDescription
         End Get
         Set(ByVal value As String)
            _SeriesDescription = Value
         End Set
      End Property

      Private _Modality As String

      <Element(DicomTag.Modality)> _
      Public Property Modality() As String
         Get
            Return _Modality
         End Get
         Set(ByVal value As String)
            _Modality = Value
         End Set
      End Property

      Private _SeriesInstanceUID As String

      <Element(DicomTag.SeriesInstanceUID)> _
      Public Property SeriesInstanceUID() As String
         Get
            Return _SeriesInstanceUID
         End Get
         Set(ByVal value As String)
            _SeriesInstanceUID = Value
         End Set
      End Property
   End Class
End Namespace
